qutebrowser/commands/runners.py

```diff
@@ -285,6 +286,14 @@ def run(self, text, count=None):
             else:
                 result.cmd.run(self._win_id, args)
 
+            mode_manager = objreg.get('mode-manager', scope='window',
+                                      window=self._win_id)
+            if (result.cmdline[0] != 'repeat-command' and
+                    result.cmd.mode_allowed(mode_manager.mode)):
```

Comment:
Is this check still needed? If I'm not missing anything, if the check isn't allowed in the current mode, `result.cmd.run` above will already bail out with a `cmdexc.PrerequisitesError`.

That means you should be able to get rid of this and `mode_allowed` above.